src/game.ts

```diff
@@ -1,115 +1,169 @@
-	store.grid = Array.from({ length: GRID_WIDTH }, () => Array.from({ length: GRID_HEIGHT }, () => ({ commitsCount: 0, intensity: 0 })));
-	store.monthLabels = Array(GRID_WIDTH).fill('');
-	let maxCommits = 1;
-
-	const now = new Date();
-	const startOfCurrentWeek = new Date(now);
-	startOfCurrentWeek.setDate(now.getDate() - now.getDay());
-
-	store.contributions.forEach((contribution) => {
-		const contributionDate = new Date(contribution.date);
-		const dayOfWeek = contributionDate.getDay();
-		const weeksAgo = Math.floor((+startOfCurrentWeek - +contributionDate) / (1000 * 60 * 60 * 24 * 7));
-
-		if (weeksAgo >= 0 && weeksAgo < GRID_WIDTH && dayOfWeek >= 0 && dayOfWeek < GRID_HEIGHT) {
-			store.grid[GRID_WIDTH - 1 - weeksAgo][dayOfWeek] = { commitsCount: contribution.count, intensity: 0 };
-			if (contribution.count > maxCommits) maxCommits = contribution.count;
-		}
-	});
-
-	for (let x = 0; x < GRID_WIDTH; x++) {
-		for (let y = 0; y < GRID_HEIGHT; y++) {
-			if (store.grid[x][y].commitsCount > 0) {
-				store.grid[x][y].intensity = store.grid[x][y].commitsCount / maxCommits;
-			}
-		}
-	}
+import { Ghost, Pacman, StoreType } from './types';
+import { fetchGitHubContributions } from './utils';
 
-	for (let x = 0; x < GRID_WIDTH; x++) {
-		const weeksAgo = GRID_WIDTH - 1 - x;
-		const columnDate = new Date(startOfCurrentWeek);
-		columnDate.setDate(columnDate.getDate() - weeksAgo * 7);
-		store.monthLabels[x] = MONTHS[columnDate.getMonth()];
-	}
+// Place pellets only on contribution cells
+const placePellets = (store: StoreType) => {
+  store.pellets = [];
+  store.contributionGrid.forEach((row, y) => {
+    row.forEach((count, x) => {
+      if (count > 0 && Math.random() > 0.7) {
+        store.pellets.push({ x, y });
+      }
+    });
+  });
 };
 
+// Place Pacman at a valid position
 const placePacman = (store: StoreType) => {
-	store.pacman = {
-		x: 0,
-		y: 0,
-		direction: 'right',
-		points: 0,
-		totalPoints: 0,
-		deadRemainingDuration: 0,
-		powerupRemainingDuration: 0,
-		recentPositions: []
-	};
-	if (store.config.outputFormat == 'canvas') Canvas.drawPacman(store);
+  const validPositions: {x: number, y: number}[] = [];
+  store.contributionGrid.forEach((row, y) => {
+    row.forEach((count, x) => {
+      if (count > 0) validPositions.push({x, y});
+    });
+  });
+
+  if (validPositions.length === 0) return;
+
+  const pos = validPositions[Math.floor(Math.random() * validPositions.length)];
+  store.pacman = {
+    x: pos.x,
+    y: pos.y,
+    direction: 'right',
+    points: 0,
+    totalPoints: 0,
+    deadRemainingDuration: 0,
+    powerupRemainingDuration: 0,
+    recentPositions: []
+  };
+
+  if (store.config.outputFormat === 'canvas') Canvas.drawPacman(store);
 };
 
+// Place ghosts at valid positions away from Pacman
 const placeGhosts = (store: StoreType) => {
-	store.ghosts = [];
-	// Center gjosts in mid grid
-	store.ghosts.push({ x: 23, y: 3, name: GHOST_NAMES[0], scared: false, target: undefined });
```

Comment:
Same thing for ghosts, they start from the middle of the game, Still we can change this via a settings.